tron/config/config_parse.py

```diff
@@ -270,6 +272,61 @@ class ValidateSecretSource(Validator):
 valid_secret_source = ValidateSecretSource()
 
 
+def valid_permission_mode(value, config_context):
+    try:
+        decimal_value = int(
+            str(value), base=8
+        )  # take in permission mode as string or int representation of an octal number. Goes from 0 to 4095 in decimal.
+    except ValueError:
+        error_msg = "Could not parse %s as octal permission mode at %s"
+        raise ConfigError(error_msg % (value, config_context.path))
+    if decimal_value > 4095 or decimal_value < 0:
+        error_msg = "Octal permission mode %s out of bound at %s"
+        raise ConfigError(error_msg % (value, config_context.path))
```

Comment:
just to make this look more consistent with modern python:
```suggestion
        error_msg = "Could not parse {} as octal permission mode at {}"
        raise ConfigError(error_msg.format(value, config_context.path))
    if decimal_value > 4095 or decimal_value < 0:
        error_msg = "Octal permission mode {} out of bound at {}"
        raise ConfigError(error_msg.format(value, config_context.path))
```